### DISCUSSION TITLE
Can speechbrain classify_batch handle multiple audio at a time?

### DISCUSSION THREAD

**Discussion by b03213135** [speechbrain/speechbrain#2096] [Q&A]
I'm using speechbrain to detect the language of several million wav audios. I want to process multiple audio(e.g. 8 .wav audio as a batch) in one classify_batch call instead of processing only one audio in one classify_batch call. The purpose is to accelerate the speed by batch processing. GPT4 suggested some code but it encountered lots of issues.
Has anybody done the same thing before? Would you please help provide the sample code for reference? Thank you!

**Comment by TParcollet** [comment]
Hi yes, it should be feasible fairly easily. If you look at the EncoderClassifier class in Pretrained/interfaces.py you will see that it contains a classify batch function. You just need to call that instead of classify file!

**Reply by b03213135** [reply]
Thanks for answering me! Is there any sample code to use multiple audio files as the input of classify_batch? I see there is sample code in interfaces.py which inputs only 1 audio file as the input of classify_batch, but I want to input a batch of audio files to it instead of single audio file.
